### PR TITLE
Fix RIC application when layout is changed

### PR DESCRIPTION
**Pull request by mvafin**
Root cause analysis: Layout for RIC was checked for target layout first, while layout change wasn't applied yet. Was initially mentioned in issue #11156.

Solution: Check source layout only.

Ticket: 82928

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names
* [x]  Transformation preserves tensor names

Validation:
* [x]  Unit tests
* [x]  Framework operation tests
* [x]  Transformation tests
* [x]  Model Optimizer IR Reader check

Documentation:
* [x]  Supported frameworks operations list
* [x]  Guide on how to convert the **public** model
* [x]  User guide update

### CODE REVIEW

File: tools/mo/openvino/tools/mo/back/preprocessing.py [hunk 1]
```diff
@@ -399,6 +393,16 @@ def apply_preprocessing(ov_function: Model, argv: argparse.Namespace):
             else:
                 prep.output(node_name).tensor().set_layout(Layout(layout_value['target_layout']))
 
+    # Apply pre-processing builder to a function
+    ov_function = prep.build()
+
+    need_reverse = 'reverse_input_channels' in argv and argv.reverse_input_channels
+    suitable_params_ric = []
+    if need_reverse:
+        suitable_params_ric = guess_source_layouts_for_reverse_channels(ov_function=ov_function,
```
Comment (nosovmik) [inline tools/mo/openvino/tools/mo/back/preprocessing.py:402]: Is it possible to fix it in 'guess_source_layout_for_reverse_channels'?
https://github.com/mvafin/openvino/blob/85e40990821739463b571954e6c3849f6fa00850/tools/mo/openvino/tools/mo/back/preprocessing.py#L265
This function shall detect 'C' channel using 'layout_values' information even if layout is not yet set to Parameter
`Additionally checks existing layouts and detects suitable inputs for reversing of input channels`

Comment (mvafin) [reply tools/mo/openvino/tools/mo/back/preprocessing.py:402]: Yes, but there is a problem of how to look for names. We need to look through all tensor names and friendly_name, I think that overhead of running preprocessing twice shouldn't be big, what do you think would be the best?

**Review by nosovmik — APPROVED** [review]
Generally, if this way is easier than modifying 'guess_source_layout_for_reverse_channels' - then I'm ok with this. It's good that PR has a test for this, so in future it can be refactored and checked for no regressions

**Review by nosovmik — APPROVED** [review]

**Comment by mbencer** [comment]
The change is continued in https://github.com/openvinotoolkit/openvino/pull/11779

**Pull request closed without merging**